vsintegration/src/FSharp.Editor/Diagnostics/DocumentDiagnosticAnalyzer.fs

```diff
@@ -108,7 +109,38 @@ type internal FSharpDocumentDiagnosticAnalyzer [<ImportingConstructor>] () =
                 for diagnostic in checkResults.Diagnostics do
                     errors.Add(diagnostic) |> ignore
 
-            if errors.Count = 0 then
+            let! unnecessaryParentheses =
+                match diagnosticType with
+                | DiagnosticsType.Semantic -> cancellableTask { return ImmutableArray.Empty }
```

Comment:
This can just be a CancellableTask.singleton so we don't allocate and then run state machine.

Reply:
Won't I eventually want to take this out of here anyway, though?

> I shoehorned the call to the analysis logic into `DocumentDiagnosticAnalyzer.fs` to test it out, but I assume we'd ultimately want to have a separate `UnnecessaryParenthesesDiagnosticAnalyzer.fs`.

> I assume I need to add `IFSharpUnnecessaryParenthesesDiagnosticAnalyzer` and `FSharpIDEDiagnosticIds.RemoveUnnecessaryParentheses` to Roslyn. Might that also be why the bulk-fix functionality is not working for this (because Roslyn is not aware that IDE0047 is supported for F#)?